mobius-core/src/main/java/com/spotify/mobius/SafeConsumer.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * -\-\-
+ * Mobius
+ * --
+ * Copyright (c) 2017-2018 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+package com.spotify.mobius;
+
+import com.spotify.mobius.functions.Consumer;
+
+/**
+ * A {@link Connection} that ensures that an inner {@link Consumer} doesn't get any values after
+ * being disposed.
+ */
+class SafeConsumer<E> implements Connection<E> {
```

Comment:
This isn't really a `Consumer`, but a `Connection`. One way to resolve it could be a `ConsumerConnection` class that turns a `Consumer` into a `Connection` and then using that as the delegate in the `SafeConnection` class. But maybe that's too much nitpicking/telescoping. If this thing needs to be a `Connection`, I think it should have a different name at least, though.

Reply:
On reading the code again, I think there should be two factory methods for `DiscardAfterDisposeConnection`, one that takes a `Consumer` and one that takes a `Connection`.

Reply:
Is that really more clear? the safe connectable looks kind of weird if you do that

Reply:
I mean if you could do either `SafeConnection.fromConnection(c)` or `SafeConnection.fromConsumer(c)`. 

And then I don't think `Safe` is particularly good as a name, because it's not clear how it is safe - nor, perhaps, to my picky mind that it is particularly 'safe' to discard things that could potentially be actual indications of a problem. But that's ground we've already covered, and I agree that the discarding is useful. Just not necessarily 'safe'.

Reply:
have a look at SafeConnectable again - i guess i could change it to be something else than "safe", but i think it's pretty clear what it does, and being able to distinguish connection and consumer makes SafeConnectable.connect quite easy to read imo.

Reply:
If you wish to keep 'Safe', I'm not going to argue strongly against that, it's not a super-big deal. :) 

I don't think you responded to the other point about calling a `Connection` (cf the class javadoc) a `Consumer`, though. Another thing that is of course more of a detail than a critical concern, but it does look a bit odd.

Reply:
do you mean I should change it to implement Consumer and Disposable instead of Connection?